component/src/main/java/org/wso2/extension/siddhi/gpl/execution/pmml/util/PMMLUtil.java

```diff
@@ -0,0 +1,51 @@
+package org.wso2.extension.siddhi.gpl.execution.pmml.util;
+
+import org.apache.log4j.Logger;
+import org.dmg.pmml.PMML;
+import org.jpmml.model.ImportFilter;
+import org.jpmml.model.JAXBUtil;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.StringReader;
+import javax.xml.bind.JAXBException;
+import javax.xml.transform.Source;
+
+/**
+ * Class implementing Pmml Model Processor.
+ */
+public class PMMLUtil {
+    private static final Logger logger = Logger.getLogger(PMMLUtil.class);
+
+    /**
+     * Unmarshal the definition and get an executable pmml model.
+     *
+     * @return pmml model
+     */
+    public static PMML unmarshal(String pmmlDefinition) {
+
+        try {
+            File pmmlFile = new File(pmmlDefinition);
+            InputSource pmmlSource;
+            Source source;
+            // if the given is a file path, read the pmml definition from the file
+            if (pmmlFile.isFile() && pmmlFile.canRead()) {
+                pmmlSource = new InputSource(new FileInputStream(pmmlFile));
+            } else {
+                // else, read from the given definition
+                pmmlSource = new InputSource(new StringReader(pmmlDefinition));
+            }
+            source = ImportFilter.apply(pmmlSource);
+            return JAXBUtil.unmarshalPMML(source);
+        } catch (SAXException | JAXBException | FileNotFoundException e) {
```

Comment:
@pcnfernando : We need to have a finally and a close() the file and other resources

Reply:
File is only a representation of a pathname, with a few methods concerning the filesystem (like exists()) and directory handling but actual streaming input and output is done elsewhere. Streams can be opened and closed, files cannot.

InputSource, we need not to close the stream explicitly, as the parser closes it.